src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+    s32 fbdemoType;
+
+    bzero(transitionCtx, sizeof(TransitionContext));
+
+    fbdemoType = -1;
+    if (transitionType & 0x40) {
+        fbdemoType = FBDEMO_WIPE3;
+    } else if ((transitionType & 0x78) == 0x20) {
+        fbdemoType = FBDEMO_WIPE4;
+    } else if (!(transitionType & 0x60)) {
+        switch (transitionType) {
+            case TRANS_TYPE_TRIFORCE:
+                fbdemoType = FBDEMO_TRIFORCE;
+                break;
+
+            case TRANS_TYPE_WIPE:
+            case TRANS_TYPE_WIPE_FAST:
+                fbdemoType = FBDEMO_WIPE1;
+                break;
+
+            case TRANS_TYPE_FADE_BLACK:
+            case TRANS_TYPE_FADE_WHITE:
+            case TRANS_TYPE_FADE_BLACK_FAST:
+            case TRANS_TYPE_FADE_WHITE_FAST:
+            case TRANS_TYPE_FADE_BLACK_SLOW:
+            case TRANS_TYPE_FADE_WHITE_SLOW:
+            case TRANS_TYPE_FADE_WHITE_CS_DELAYED:
+            case TRANS_TYPE_FADE_WHITE_INSTANT:
+            case TRANS_TYPE_FADE_GREEN:
+            case TRANS_TYPE_FADE_BLUE:
+                fbdemoType = FBDEMO_FADE;
+                break;
+
+            case TRANS_TYPE_FILL_WHITE_FAST:
+            case TRANS_TYPE_FILL_WHITE:
+                this->transitionMode = TRANS_MODE_FILL_WHITE_INIT;
+                break;
+
+            case TRANS_TYPE_INSTANT:
+                this->transitionMode = TRANS_MODE_INSTANT;
+                break;
+
+            case TRANS_TYPE_FILL_BROWN:
+                this->transitionMode = TRANS_MODE_FILL_BROWN_INIT;
+                break;
+
+            case TRANS_TYPE_SANDSTORM_PERSIST:
+                this->transitionMode = TRANS_MODE_SANDSTORM_INIT;
+                break;
+
+            case TRANS_TYPE_SANDSTORM_END:
+                this->transitionMode = TRANS_MODE_SANDSTORM_END_INIT;
+                break;
+
+            case TRANS_TYPE_CS_BLACK_FILL:
+                this->transitionMode = TRANS_MODE_CS_BLACK_FILL_INIT;
+                break;
+
+            case TRANS_TYPE_CIRCLE:
+                fbdemoType = FBDEMO_CIRCLE;
+                break;
+
+            case TRANS_TYPE_WIPE5:
+                fbdemoType = FBDEMO_WIPE5;
+                break;
+
+            default:
+                fbdemoType = -1;
+                __assert("../z_play.c", 1420);
+        }
+    } else {
+        fbdemoType = -1;
+        __assert("../z_play.c", 1423);
+    }
+
+    transitionCtx->transitionType = transitionType;
+    transitionCtx->fbdemoType = fbdemoType;
+    if (fbdemoType != -1) {
+        Transition_Init(transitionCtx);
+    }
+}
+
+const char D_801DFA18[] = "来た!!!!!!!!!!!!!!!!!!!!!";
+const char D_801DFA34[] = "all";
+const char D_801DFA38[] = "a";
+const char D_801DFA3C[] = "a";
+const char D_801DFA40[] = "b";
+const char D_801DFA44[] = "b";
+const char D_801DFA48[] = "c";
+const char D_801DFA4C[] = "c";
+const char D_801DFA50[] = "d";
+const char D_801DFA54[] = "d";
+const char D_801DFA58[] = "e";
```

Comment:
What do we do with this mess? Just leave it here? Don't suppose this can be made into an array to compact it somehow

Reply:
Alright yeah, played around with this a bit and this matches:
```c
const char D_801DFA18[] = "来た!!!!!!!!!!!!!!!!!!!!!";
const char D_801DFA34[][4] = {
    "all", "a",  "a",  "b",  "b",  "c",  "c",  "d",   "d",   "e",  "e",  "f",  "fa", "fa", "fb", "fb",
    "fc",  "fc", "fd", "fd", "fe", "fe", "fg", "fg",  "fh",  "fh", "fi", "fi", "fj", "fj", "fk", "fk",
    "f",   "g",  "g",  "h",  "h",  "i",  "i",  "all", "all", "a",  "b",  "c",  "d",  "e",  "f",  "g",
    "h",   "i",  "f",  "fa", "fb", "fc", "fd", "fe",  "ff",  "fg", "fh", "fi", "fj", "fk",
};
```
so that's probably better

Reply:
does debug say what this is for

Reply:
Looks like `"来た!!!!!!!!!!!!!!!!!!!!!"` can be inferred from OoT:

```c
case TRANS_MODE_SANDSTORM_END_INIT:
    if (this->transitionTrigger == TRANS_TRIGGER_END) {
        this->envCtx.sandstormState = SANDSTORM_DISSIPATE;
        this->envCtx.sandstormPrimA = 255;
        this->envCtx.sandstormEnvA = 255;
        // "It's here!!!!!!!!!"
        LOG_STRING("来た!!!!!!!!!!!!!!!!!!!!!", "../z_play.c", 3471);
        this->transitionMode = TRANS_MODE_SANDSTORM_END;
    } else {
        this->transitionMode = TRANS_MODE_SANDSTORM_INIT;
    }
    break;
```

I'm not seeing these letter string in debug at a first glance (at least what's been put into `z_play`, the things I'm seeing are:

```c
((void)"kankyo changed water, sound on\n");

((void)"kankyo changed water off, sound off\n");

// Sound initials are here. 111
((void)"\n\n\nサウンドイニシャル来ました。111\n");

((void)"\nSEQ_ID_TIME_LIMIT=[%d] Na_CheckPlayingBgmFlag = [%d]");

// Sound initials are here.
((void)"\n\n\nサウンドイニシャル来ました。\n");

// The sound initials for hijacking BGM have arrived.
((void)"\n\n\n乗っ取りＢＧＭ用サウンドイニシャル来ました。\n");

((void)"来た!!!!!!!!!!!!!!!!!!!!!");

// Call fbdemo_init failed!
((void)"fbdemo_init呼出し失敗！\n");
```

`fbdemo_init` based on OoT is `TransitionUnk_Init`
```c
TransitionUnk* TransitionUnk_Init(TransitionUnk* this, s32 row, s32 col) {
    osSyncPrintf("fbdemo_init(%08x, %d, %d)\n", this, row, col);
```
So in MM, that string is probably somewhere around here:
```c
if (TransitionUnk_Init(&sTrnsnUnk, 10, 7) == NULL) {
    gTrnsnUnkState = 0;
} else {
    D_801F6D0C = gZBufferPtr;
    gTrnsnUnkState = 3;
    Game_SetFramerateDivisor(&this->state, 1);
}
```

But nothing on those batch of letters. Maybe those are related to all transition types?